universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -44,4 +47,36 @@ public Program updateProgramSync(Program program) {
     ebeanServer.update(program);
     return program;
   }
+
+  public CompletionStage<Void> publishProgramAsync(Program program) {
+    return supplyAsync(
+        () -> {
+          try {
+            ebeanServer.beginTransaction();
+            program.setLifecycleStage(LifecycleStage.ACTIVE);
+            List<Program> allOldPrograms =
+                ebeanServer
+                    .find(Program.class)
+                    .where()
+                    .eq("name", program.getProgramDefinition().name())
+                    .eq("lifecycle_stage", LifecycleStage.ACTIVE)
+                    .not()
+                    .eq("id", program.id)
+                    .findList();
+            for (Program oldProgram : allOldPrograms) {
+              oldProgram.setLifecycleStage(LifecycleStage.OBSOLETE);
+              oldProgram.save();
+            }
+            program.save();
+            ebeanServer.commitTransaction();
+            return null;
+          } finally {
+            ebeanServer.endTransaction();
+          }
+        });
```

Comment:
Suggestion (blocking): 
```suggestion
        }, executionContext);
```